docs/Migration_Guide_Zephyr.md

```diff
@@ -0,0 +1,587 @@
+Golioth-specific Kconfig symbols. You may type `?` when viewing a symbol
+to see the symbol name and more information about how it is selected.
+
+### 3. Sample library Kconfig symbols
+
+If you are using the sample library in your application, change the main
+symbol to the updated name which uses the singular `SAMPLE`:
+
+```config
+/* Remove this from your existing code */
+CONFIG_GOLIOTH_SAMPLES_COMMON=y
+
+/* Replace it with this updated syntax */
+CONFIG_GOLIOTH_SAMPLE_COMMON=y
+```
+
+> :warning: **The symbol name has changed**: The symbol used to enable the
+Golioth sample library has dropped the `S` from the end of `SAMPLES`.
+
+Other symbol changes/additions in the Sample library are less commonly
+used by our customers and  beyond the scope of this guide. Please review
+the Golioth example code for guidance.
+
+## Update header file includes
+
+Update the Golioth header files included in your project's source code.
+
+All Golioth Firmware SDK header files are now prefixed with `golioth`. A
+complete [list of the available header
+files](https://github.com/golioth/golioth-firmware-sdk/tree/main/include/golioth)
+is available in the top-level `include` directory.
+
+For example, you will always need to include the Golioth client in your
+builds, here is a comparison of the old and new naming:
+
+```c
+/* Remove this from your existing code */
+#include <net/golioth/system_client.h>
+
+/* Replace it with this updated syntax */
+#include <golioth/client.h>
+```
+
+The naming scheme for these files has been updated so that it is easier
+to understand what includes do. This comes with the added benefit of
+making it easier to intuitively know what header file you need.
+
+### Sample header files
+
+Golioth example code uses a common sample library. This is not
+technically a part of the core SDK (it servers as a guide for developing
+your own applications based on the SDK). Header files for sample library
+largely retain the same naming scheme, though several new headers have
+been added.
+
+```c
+/* This include remains unchanged */
+#include <samples/common/net_connect.h>
+```
+
+## Change Golioth API calls
+
+Most of the API names and formats have changed. The majority of the
+migration work will be in correctly updating these calls. This section
+highlights the most commonly needed changes.
+
+> :information_source: **Compare Sample Code**: The most direct
+> comparison of these changes is found by comparing samples between the
+> deprecated [Golioth Zephyr
+> SDK](https://github.com/golioth/golioth-zephyr-sdk/tree/main/samples)
+> and the [Golioth Firmware
+> SDK](https://github.com/golioth/golioth-firmware-sdk/tree/main/examples/zephyr)
+> that replaces it.
+
+### 1. Creating the Golioth client
+
+- The Golioth client now requires a `golioth_client_config` struct that
+  contains a credentials for authenticating with Golioth. This is used
+  for either certificate authentication or pre-shared key
+  authentication.
```

Comment:
My suggested wording might need work, but we should provide information about how to use the credential tag functionality that Marcin is adding to enable offloaded DTLS sockets
```suggestion
  for either certificate authentication, pre-shared key
  authentication, or credential tag authentication.
```

Reply:
It should be in the migration guide. For those who were provisioning credentials in a custom way, especially those using offloaded DTLS sockets on the nrf9160, this is the information they need to move to the CP SDK.

Reply:
I think adding another example of the struct when using tags, like we have for psk and pki is enough. We don't need the full explanation in this guide.

Reply:
Okay, I've added an example config.